### PR TITLE
fix: Kills recursively processes from spawned agent shell commands

### PR DESCRIPTION
**Pull request by jsibbison-square**
This adds extra targetting on the bash shell command from the developer system. Discovered that depending on the type of shell command executed the existing 'kill_on_drop' technique would only kill the original child_process not any grandchildren, resulting in the processes continuing.

This could be done by creating a script in a file that appends every second to a file, then running it from goose. A cancel event would not stop the file running.

The implementation to work around this uses a singleton to store running processes and [kill_tree](https://github.com/oneofthezombies/kill-tree) crate to recursively kill its processes when a 'cancel' comes in. Tested in cli and gui.

### Caveat
The only situation I've run into so far that doesn't work is when goose runs a process in the background (background process will be the grandchild), It's child pid closes (its finished triggering a background process) and the background process gets re-owned by pid 1 so we have no way to find it. eg. `bash script.sh &`

### CODE REVIEW

**Comment by michaelneale** [comment]
I don't think "The only situation I've run into so far that doesn't work is when goose runs a process in the background (background process will be the grandchild)" is too much of an issue in practice would it be?

**Comment by michaelneale** [comment]
I wonder if https://users.rust-lang.org/t/killing-subprocesses-of-std-command/117905 is the more robust way (ie will always have a relevant parent/pid set?)

File: crates/goose/src/process_store.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+/// Kill all stored processes
+pub fn kill_processes() {
+    let mut killed_processes = Vec::new();
+    {
+        let store = PROCESS_STORE.lock().unwrap();
+        for &pid in store.iter() {
+            let config = Config {
+                signal: "SIGKILL".to_string(),
```
Comment (michaelneale) [inline crates/goose/src/process_store.rs:33]: brutal. I like it

**Comment by michaelneale** [comment]
@jsibbison-square fiddly to test - worth asking goose to make a test case for it spinning up things? but LGTM

**Review by baxen — APPROVED** [review]
this makes sense. it increasingly pushes us towards a single goosed per connection, but that seems fine and we could always refactor if needed. 

👍

**Comment by michaelneale** [comment]
@baxen yes we are at I believe one goosed per connection effectively anyway by design IIUC.

**Comment by jsibbison-square** [comment]
Test added, thankfully it catches that Command.kill does not get rid of the child whereas this kill_tree implementation does 👍 .

**Pull request merged**